contents/docs/product-analytics/troubleshooting.mdx

```diff
@@ -63,49 +63,51 @@ posthog = PostHog::Client.new({
 
-### 2. There is a bug within the library being used.
+### 2. There is a bug the library
+
+Sometimes, bugs in the PostHog library may be the cause of missing events. However, this is quite uncommon.
 
-Certain events could fail to reach PostHog if there is a bug in the used library. (this is quite uncommon)
+To debug if this is the issue, most of our SDKs have an option to enable debug logging. Consult the relevant [SDK docs](/docs/libraries) for how to do this.
 
-#### Web libraries
-1. Open the devtools and switch to the network tab.
-2. Look for the failed requests to app.posthog.com/e (or eu.posthog.com/e if using the EU Cloud).
+If you're using the [JavaScript web](/docs/libraries/js) library, you can identify failed requests using the network tab in your browser devtools:
+
+1. Open devtools in your browser and switch to the network tab.
+2. Identify the failed requests to app.posthog.com/e (or eu.posthog.com/e if you're using EU Cloud).
 3. Right click one of those request and click Copy -> Copy as cURL.
+4. See below on how to [report your issue](/docs/product-analytics/troubleshooting#report-your-specific-issue) and attach your cURL command to get further support.
 
 ![network request being sent to posthog](../../images/docs/product-analytics/network-request-to-posthog.png)
 
-3. See how to [Report your specific issue](/docs/product-analytics/troubleshooting#report-your-specific-issue) to get further support.
+### 3. Events are not ingested after being received by PostHog.
 
-#### Other libraries
-See the respective debugging guides, e.g. [Node.js debugging](https://posthog.com/docs/libraries/node#debugging-and-exceptions).
+If your library is configured correctly and successfully sending events to PostHog, events may get lost in the ingestion pipeline – e.g. if there is an [active incident](https://status.posthog.com).
 
-### 3. Events are not ingested after being received by PostHog.
+To confirm if this is the issue, use the following steps:
+
+1. Ensure that PostHog is not [experiencing an incident](https://status.posthog.com), as this may cause ingestion lags.
 
-If the library is configured correctly and successfully sending events to PostHog. Events could get lost in the ingestion pipeline e.g. when there is an [active incident](https://status.posthog.com).
+2. Send your event as a cURL request directly
+    - i) Copy your cURL request.
+         - If you're using the JavaScript web library, you can copy the cURL request from the network tab in your browser devtools.
+         - If you're using any other library, you can track your outgoing network requests using something like [Wireshark](https://www.wireshark.org/) and format them as a cURL request.
+    - ii) [Decode your request](/docs/product-analytics/troubleshooting#step-1-decode-your-curl-command).
+    - iii) Send your event a cURL request. Wait 1 hour and check if the event is still not visible in PostHog.
 
```

Comment:
lets add another step here: check ingestion warnings

Reply:
Sounds good. How do you check ingestion warnings?

Reply:
in the app data management -> ingestion warnings